sup3r/models/with_obs.py

```diff
@@ -0,0 +1,254 @@
+"""Sup3r model with training on observation data."""
+
+import logging
+
+import numpy as np
+import tensorflow as tf
+
+from sup3r.utilities.utilities import RANDOM_GENERATOR
+
+from .base import Sup3rGan
+
+logger = logging.getLogger(__name__)
+
+
+class Sup3rGanWithObs(Sup3rGan):
+    """Sup3r GAN model which includes mid network observation fusion. This
+    model is useful for when production runs will be over a domain for which
+    observation data is available."""
```

Comment:
Somewhere, probably in this header, can you describe the nuances of training with (fake)observations vs. inference with real observations? I think that functions described with "obs for current batch" are for fake observations from training data but it's not really clear. 

I also want to know how someone would pass in real observations at inference time. You don't subclass the generate() function here so i don't see anywhere that would describe how to provide real observations in a fwp.

Reply:
Yeah thanks for this feedback. Hard to get this perspective when you're "in it" for a while. Obs are used in fwp the same way as other exogenous data sources, but the ObsRasterizer should be specified (this allows NaNs to make it through the rasterization) or obs feature names should include the suffix "_obs" to automatically use the ObsRasterizer.